Docs with non-coerced JSON value can not be filtered on. Panic.

Setup:
```shell
$ defradb client schema add 'type User { name: String, custom: JSON }'
$ defradb client collection create '{ "name": "Fred", "custom": "{\"age\": 20}"}'\
```
Check current state:
```graphql
query {
  User {
    name
    custom
  }
}
```
Response:
```json
{
  "data": {
    "User": [
      {
        "custom": "{\"age\": 20}",
        "name": "Fred"
      }
    ]
  }
}
```
Send a request with filter:
```graphql
query {
  User(filter: {custom: {age: {_gt: 10}}}) {
    name
    custom
  }
}
```
Produces panic:
```shell
panic: interface conversion: interface {} is string, not map[string]interface {}

 -> github.com/sourcenetwork/defradb/internal/planner/mapper.(*ObjectProperty).GetProp
 ->   github.com/sourcenetwork/defradb/internal/planner/mapper/targetable.go:88

    github.com/sourcenetwork/defradb/internal/connor.eq
      github.com/sourcenetwork/defradb/internal/connor/eq.go:37
    github.com/sourcenetwork/defradb/internal/connor.matchWith
      github.com/sourcenetwork/defradb/internal/connor/connor.go:63
    github.com/sourcenetwork/defradb/internal/connor.eq
      github.com/sourcenetwork/defradb/internal/connor/eq.go:37
    github.com/sourcenetwork/defradb/internal/connor.Match
      github.com/sourcenetwork/defradb/internal/connor/connor.go:48
    github.com/sourcenetwork/defradb/internal/planner/mapper.RunFilter
      github.com/sourcenetwork/defradb/internal/planner/mapper/mapper.go:1482
    github.com/sourcenetwork/defradb/internal/db/fetcher.(*DocumentFetcher).fetchNext
      github.com/sourcenetwork/defradb/internal/db/fetcher/fetcher.go:618
    github.com/sourcenetwork/defradb/internal/db/fetcher.(*DocumentFetcher).FetchNext
      github.com/sourcenetwork/defradb/internal/db/fetcher/fetcher.go:578
    github.com/sourcenetwork/defradb/internal/lens.(*lensedFetcher).FetchNext
      github.com/sourcenetwork/defradb/internal/lens/fetcher.go:134
    github.com/sourcenetwork/defradb/internal/planner.(*scanNode).Next
      github.com/sourcenetwork/defradb/internal/planner/scan.go:227
    github.com/sourcenetwork/defradb/internal/planner.(*selectNode).Next
      github.com/sourcenetwork/defradb/internal/planner/select.go:153
    github.com/sourcenetwork/defradb/internal/planner.(*selectTopNode).Next
      github.com/sourcenetwork/defradb/internal/planner/select.go:70
    github.com/sourcenetwork/defradb/internal/planner.(*operationNode).Next
      github.com/sourcenetwork/defradb/internal/planner/operation.go:107
    github.com/sourcenetwork/defradb/internal/planner.(*Planner).executeRequest
      github.com/sourcenetwork/defradb/internal/planner/planner.go:482
    github.com/sourcenetwork/defradb/internal/planner.(*Planner).RunRequest
...
```

Looks like we need to drop support for non-coerced json all together, otherwise we will have to handle both versions.

Another point is that `@default` directive accepts only non-coerced version, which means that all docs with json values set by default will produce the same result.

--- a/internal/connor/connor.go
+++ b/internal/connor/connor.go
@@ -1,8 +1,11 @@
-/*
-Package connor provides a domain-specific language to express conditions against data.
-
-It is derived from https://github.com/SierraSoftworks/connor.
-*/
+// Package connor provides a domain-specific language to express conditions against data.
+//
+// It is derived from https://github.com/SierraSoftworks/connor.
+//
+// Note to developers:
+// Never assume that the data given to an operator is of a certain type.
+// Operators should work with any data type so that untyped data
+// such as JSON can be filtered as expected.
 package connor
 
 const (

--- a/internal/connor/eq.go
+++ b/internal/connor/eq.go
@@ -75,8 +75,8 @@ func objectsEqual(condition map[string]any, data any) (bool, error) {
 	if data == nil {
 		return condition == nil, nil
 	}
-	d := data.(map[string]any)
-	if len(d) != len(condition) {
+	d, ok := data.(map[string]any)
+	if !ok || len(d) != len(condition) {
 		return false, nil
 	}
 	for k, v := range d {

--- a/tests/integration/query/json/with_eq_test.go
+++ b/tests/integration/query/json/with_eq_test.go
@@ -162,3 +162,65 @@ func TestQueryJSON_WithEqualFilterWithNullValue_ShouldFilter(t *testing.T) {
 
 	testUtils.ExecuteTestCase(t, test)
 }
+
+func TestQueryJSON_WithEqualFilterWithAllTypes_ShouldFilter(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple query with JSON _eq filter all types",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						Name: String
+						Custom: JSON
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Shahzad",
+					"Custom": "32"
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Andy",
+					"Custom": [1, 2]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Fred",
+					"Custom": {"one": 1}
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "John",
+					"Custom": false
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "David",
+					"Custom": 32
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {Custom: {_eq: {one: 1}}}) {
+						Name
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"Name": "Fred",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}

